client/jira.go

```diff
@@ -140,9 +140,14 @@ func GetIssueInfo(issueId string) (domain.Issue, error) {
 	return issue, nil
 }
 
-func CreateIssue(project, issueType, title string) (domain.Issue, error) {
+func CreateIssue(project, issueType, summary string) (domain.Issue, error) {
 	var issue domain.Issue
-	issueRequest := newCreateIssueRequest(project, issueType, title)
+	issueRequest := IssueRequest{
+		Fields: RequestFields{
+			Project:   ProjectRequest{Key: project},
```

Comment:
Just had another look at this and I think we can simplify this a bit so it's not so much boilerplate.

Currently, we have a lot of nested structures:
```
IssueRequest
     \ RequestFields
            \ ProjectRequest
                 \ Key
            \ IssueTypeRequest
                  \ Name
```

I think it would be great to cut down on all of this by [embedding nested structs](https://travix.io/type-embedding-in-go-ba40dd4264df).

Here is my suggestion, embed `ProjectRequest` and `IssueTypeRequest` in `RequestFields` and then embed `RequestFields` in `IssueRequest`. The final result will be like this:
```go
type IssueRequest struct {
   RequestFields `json:"fields"`
}

type RequestFields struct {
    ProjectRequest `json:"project"
    ...
    ...
}

func CreateIssue() {
   req := IssueRequest{
      ProjectRequest{project},
      Summary: summary,
      IssueTypeRequest{issueType},
  }
```

I'm not sure but you might not even have to specify `ProjectRequest` and may be able to just use `{}` have a look!

Reply:
And why everything is postfixed with `Request`?

Reply:
@pamelin Idk - I didn't write this command. I fixed a minor issue and made it work!